### PR TITLE
Refactor README and add a mortgage calculator demo

### LINKED ISSUE #1380: Mortgage calculator as a new code example in readme
### Description

Prepare a demo of HF used as a calculator for named variables for the README. So instead of having this example (https://github.com/handsontable/hyperformula?tab=readme-ov-file#installation-and-usage) we could have sth more like a mortgage calculator where you pass consts for the PMT [function](https://support.microsoft.com/en-us/office/using-excel-formulas-to-figure-out-payments-and-savings-11cb708f-c137-4ef8-bcf3-5137aaeb4b20#:~:text=%3DPMT(5%25%2F12%2C,present%20value%20of%20the%20loan).). 

### Links

- https://github.com/handsontable/hyperformula/issues/1255
- https://handsoncode.slack.com/archives/C031GBX62SE/p1708427285153139 (internal discussion)

### PR DESCRIPTION
**Pull request by sequba**
### Context

New demo for the README. Using PMT function to implement a simple mortgage calculator.

Should be merged together with https://github.com/handsontable/hyperformula-demos/pull/24

### How did you test your changes?

- [x] verify that the codesandbox link works
- [x] verify that the code snippet works when copied to codesandbox

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [ ] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [x] Change to the documentation

### Related issues:
Fixes #1380

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to HyperFormula](https://hyperformula.handsontable.com/guide/contributing.html) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2).
- [x] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets.
- [ ] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,90 +22,89 @@
...
+// Add an empty sheet
+const sheetName = hf.addSheet('Mortgage Calculator');
+const sheetId = hf.getSheetId(sheetName);
 
-// build an instance with defined options and data
-const hfInstance = HyperFormula.buildFromArray(data, options);
+// Enter the mortgage parameters
+hf.addNamedExpression('AnnualInterestRate', '8%');
+hf.addNamedExpression('NumberOfMonths', 10);
+hf.addNamedExpression('LoanAmount', 10000);
```
Comment (evanSe) [inline README.md:90]: Lets do something more realistic 
```suggestion
hf.addNamedExpression('AnnualInterestRate', '8%');
hf.addNamedExpression('NumberOfMonths', 360);
hf.addNamedExpression('LoanAmount', 800000);
```

**Review by evanSe — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -22,90 +22,89 @@
 
 ---
 
-HyperFormula is a headless spreadsheet built on top of TypeScript. It is a parser and evaluator of Excel formulas for web applications. You can use it in a browser or as a service, with Node.js as your back-end technology.
-- High-speed Excel formula parsing and evaluating
-- A library of nearly [400 built-in functions](https://hyperformula.handsontable.com/guide/built-in-functions.html)
-- Support for [internationalization](https://hyperformula.handsontable.com/guide/i18n-features.html) with 17 built-in languages
-- Support for [custom functions](https://hyperformula.handsontable.com/guide/custom-functions.html)
+HyperFormula is a headless spreadsheet built in TypeScript, serving as both a parser and evaluator of spreadsheet formulas. It can be seamlessly integrated into your browser or utilized as a service with Node.js as your back-end technology.
```
Comment (krzysztofspilka) [inline README.md:25]: Let's remove a word `seamlessly`

File: README.md [hunk 3]
```diff
@@ -22,90 +22,89 @@
...
+
+## What HyperFormula can be used for?
+HyperFormula doesn't assume any existing user interface, making it a great general-purpose library that can be used in various business applications. Here are some examples:
```
Comment (krzysztofspilka) [inline README.md:28]: Let's remove a word `great`

File: README.md [hunk 4]
```diff
@@ -22,90 +22,89 @@
...
+- Educational app
+- Online calculator
+
+## Features
+
 - Function syntax compatible with [Microsoft Excel](https://hyperformula.handsontable.com/guide/compatibility-with-microsoft-excel.html) and [Google Sheets](https://hyperformula.handsontable.com/guide/compatibility-with-google-sheets.html)
+- High-speed parsing and evaluation of spreadsheet formulas
+- A library of [~400 built-in functions](https://hyperformula.handsontable.com/guide/built-in-functions.html)
+- Support for [custom functions](https://hyperformula.handsontable.com/guide/custom-functions.html)
 - [Support for Node.js](https://hyperformula.handsontable.com/guide/server-side-installation.html#install-with-npm-or-yarn)
```
Comment (krzysztofspilka) [inline README.md:42]: Let's make all the links on the list like this [Support for Node.js](https://hyperformula.handsontable.com/guide/server-side-installation.html#install-with-npm-or-yarn) - the entire list item should be wrapped in a link.

File: README.md [hunk 5]
```diff
@@ -22,90 +22,89 @@
...
 - Support for [CRUD operations](https://hyperformula.handsontable.com/guide/basic-operations.html)
 - Support for [clipboard](https://hyperformula.handsontable.com/guide/clipboard-operations.html)
 - Support for [named expressions](https://hyperformula.handsontable.com/guide/named-expressions.html)
 - Support for [data sorting](https://hyperformula.handsontable.com/guide/sorting-data.html)
-- Support for [React](https://hyperformula.handsontable.com/guide/integration-with-react.html), [Angular](https://hyperformula.handsontable.com/guide/integration-with-angular.html), and [Vue.js](https://hyperformula.handsontable.com/guide/integration-with-vue.html)
-- Open-source license
-- Actively maintained by the team that stands behind [Handsontable - JavaScript Data Grid](https://handsontable.com/)
+- Support for [formula localization](https://hyperformula.handsontable.com/guide/i18n-features.html) with 17 built-in languages
+- GPLv3 license
+- Maintained by the company that stands behind the [Handsontable](https://handsontable.com/) data grid
```
Comment (krzysztofspilka) [inline README.md:50]: by the company -> by the team

File: README.md [hunk 6]
```diff
@@ -22,90 +22,89 @@
...
-- Smart documents
-- Educational apps
-- Online calculators
+[Run this code in CodeSandbox](https://codesandbox.io/p/sandbox/github/handsontable/hyperformula-demos/tree/feature/issue-1380/mortgage-calculator)
 
 ## Contributing
 
-Help us build the fastest and most flexible calculation engine for
-business web apps. Please read the [Contributing Guide](https://hyperformula.handsontable.com/guide/contributing.html) before making a pull request.
+Please read the [Contributing Guide](https://hyperformula.handsontable.com/guide/contributing.html) before making a pull request.
```
Comment (krzysztofspilka) [inline README.md:103]: Better: Contributions are welcome, but before you make them, please read the [Contributing Guide]() and accept the Contributor License Agreement.

File: README.md [hunk 7]
```diff
@@ -22,90 +22,89 @@
...
 ## License
 
-HyperFormula is available under the open source license ([GPLv3](https://github.com/handsontable/hyperformula/blob/master/LICENSE.txt)).
-
-To buy a commercial license, please write to us at sales@handsontable.com
+HyperFormula is available under two different licenses: GPLv3 and commercial. The commercial license can be purchased by [contacting our team](https://handsontable.com/get-a-quote) at Handsontable.
 
 ## Copyrights
-© 2023 [Handsoncode](https://handsontable.com)
+© 2024 Handsoncode
```
Comment (krzysztofspilka) [inline README.md:110]: Better: Copyright (c) Handsoncode

**Review by krzysztofspilka — CHANGES_REQUESTED** [review]

File: README.md
Comment (sequba) [inline README.md:1]: TODO: copy the content of README.md to the docs main page docs/index.md.

File: README.md [hunk 8]
```diff
@@ -22,90 +22,89 @@
...
-HyperFormula doesn't assume any existing user interface, making it a great general-purpose library that can be used in various business applications. Here are some examples:
-
-- Spreadsheets
-- Business logic builders
-- Forms and form builders
-- Computation notebooks
-- Smart documents
-- Educational apps
-- Online calculators
+[Run this code in CodeSandbox](https://codesandbox.io/p/sandbox/github/handsontable/hyperformula-demos/tree/feature/issue-1380/mortgage-calculator)
```
Comment (sequba) [inline README.md:100]: TODO: before merging update the codesandbox link to the production one

**Review by evanSe — APPROVED** [review]

**Comment by sequba** [comment]
@krzysztofspilka @evanSe The mortgage calculator demo in codesandbox should be fixed now.

File: README.md
Comment (sequba) [reply README.md:1]: done

File: README.md (hunk 8 above)
Comment (sequba) [reply README.md]: done

**Pull request merged**